apis/Google.Cloud.PubSub.V1/Google.Cloud.PubSub.V1.Tests/SubscriberClientTest.cs

```diff
@@ -1111,5 +1119,103 @@ public void AckModifyAckDeadlineFault_Thrown([CombinatorialValues(true, false)]
                 Assert.Equal(1, fake.ClientShutdowns.Count);
             });
         }
+
+        [Theory, CombinatorialData]
+        public void ExactlyOnce_TemporaryFault([CombinatorialValues(true, false, null)] bool? ackNackOrExtends)
+        {
+            var msgs = new[]
+            {
+                ServerAction.Data(TimeSpan.Zero, new[] { "1" }),
+                ServerAction.Data(TimeSpan.Zero, new[] { "2" }),
+                ServerAction.Data(TimeSpan.Zero, new[] { "3" }),
+                ServerAction.Data(TimeSpan.Zero, new[] { "4" }),
+                ServerAction.Inf()
+            };
+
+            // Any temporary failure in ack/nack/extend request should not be thrown to the client. It should be retried.
+            var exception = new RpcException(new Status(StatusCode.DeadlineExceeded, ""), "");
```

Comment:
So this is a *general* exception rather than one applying to specific messages, yes? I assume we will also end up with tests where some succeed, some fail permanently and some fail temporarily?

Reply:
Right Jon. I will add them.